pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -726,12 +730,29 @@ func getOSDInfo(d *apps.Deployment) ([]OSDInfo, error) {
 		}
 		locationPrefix := "--crush-location="
 		if strings.HasPrefix(a, locationPrefix) {
+			locationFound = true
 			// Extract the same CRUSH location as originally determined by the OSD prepare pod
 			// by cutting off the prefix: --crush-location=
 			osd.Location = a[len(locationPrefix):]
 		}
 	}
 
+	if !locationFound {
+		pods, err := c.context.Clientset.CoreV1().Pods(d.Namespace).List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=%d", OsdIdLabelKey, osd.ID)})
+		if err != nil {
+			logger.Errorf("error getting pods %v", err)
```

Comment:
You might want to split this block out to a new method so you can easily return when `err != nil`. If you log the error and continue, it might crash. If we do hit an error we can just log it and skip adding the location here, instead of failing the whole osd. This is just a best effort to find the location.